src/plugins/kibana/public/settings/sections/indices/add_data_steps/pattern_review_step/pattern_review_step.js

```diff
@@ -24,35 +26,51 @@ modules.get('apps/settings')
       controllerAs: 'reviewStep',
       bindToController: true,
       controller: function ($scope, Private) {
+        this.errors = [];
         if (_.isUndefined(this.indexPattern)) {
           this.indexPattern = {};
         }
 
         const knownFieldTypes = {};
         this.dateFields = [];
-        this.pipeline.model.processors.forEach((processor) => {
-          if (processor.typeId === 'geoip') {
-            const field = processor.targetField || 'geoip';
-            knownFieldTypes[field] = 'geo_point';
+        if (this.pipeline) {
+          this.pipeline.model.processors.forEach((processor) => {
+            if (processor.typeId === 'date') {
+              const field = processor.targetField || '@timestamp';
+              knownFieldTypes[field] = 'date';
+              this.dateFields.push(field);
+            }
+          });
+        }
+
+        const fields = {};
+        forEachField(this.sampleDoc, (value, fieldName) => {
+          let type = knownFieldTypes[fieldName] || typeof value;
+          if (isGeoPointObject(value)) {
+            type = 'geo_point';
+            knownFieldTypes[fieldName] = 'geo_point';
           }
-          else if (processor.typeId === 'date') {
-            const field = processor.targetField || '@timestamp';
-            knownFieldTypes[field] = 'date';
-            this.dateFields.push(field);
+
+          if (!_.isUndefined(fields[fieldName]) && (fields[fieldName].type !== type)) {
+            this.errors.push(`Error in field ${fieldName} - conflicting types '${fields[fieldName].type}' and '${type}'`);
+          }
+          else {
+            fields[fieldName] = {type, value};
           }
         });
 
+        const indexPatternFields = _(fields)
+        .map((field, fieldName) => {
+          return {name: fieldName, type: field.type};
+        })
+        .reject({type: 'object'})
+        .value();
+
         _.defaults(this.indexPattern, {
           id: 'filebeat-*',
           title: 'filebeat-*',
           timeFieldName: pickDefaultTimeFieldName(this.dateFields),
```

Comment:
Does this need to be initialized here? If you change the way that `isTimeBased` is initialized, you could rely on the initial firing of the `reviewStep.isTimeBased` watcher to set the value of `reviewStep.timeFieldName`.